internal/controller/entra/main.go

```diff
@@ -134,41 +143,59 @@ func (c *Controller) IsStaffMember(ctx context.Context, username types.Username)
 
 func (c *Controller) SendInvite(ctx context.Context, email string, sponsor types.Sponsor) error {
```

Comment:
this function is too complex – I think should be split on the switch if the users exists in entra or not. e.g.

```go
func (c *Controller) SendInvite(ctx context.Context, email string, sponsor types.Sponsor) error {
	user, err := c.userData(ctx, types.Username(email))
	if err == nil {
		return c.sendInviteExistingEntraUser(...)
	} else if errors.Is(err, types.ErrNotFound) {
		return c.sendInviteNewEntraUser(...)
	} else {
		return err
	}
}
```